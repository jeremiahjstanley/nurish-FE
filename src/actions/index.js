export const addFormulas = (formulas) => ({
  type: 'ADD_FORMULAS',
  formulas
})

export const formulaFetchDataSuccess = (formulas) => ({
  type: 'FORMULA_FETCH_DATA_SUCCESS',
  formulas
})

export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})

export const hasErrored = (bool) => ({
  type: 'HAS_ERRORED',
  hasErrored: bool
})