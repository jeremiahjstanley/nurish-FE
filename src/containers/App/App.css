.app {
  text-align: center; }

.app .links {
  animation: shrink-form 1s;
  width: 25%; }

@keyframes shrink-form {
  0% {
    width: 45%; }
  100% {
    width: 25%; } }

.form-container {
  background-color: rgba(255, 255, 255, 0.2);
  border-radius: 30px;
  height: 25vh;
  margin: 0 auto;
  min-width: 20rem;
  padding: 1.5rem;
  text-align: center;
  transition: width 1 ease;
  width: 25%; }

.links {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly; }

.links a {
  width: 75%; }

.link-container {
  display: flex;
  justify-content: space-between; }

.link-container a {
  max-width: 10rem;
  width: 30%; }

a {
  border-radius: 30px;
  border: 1px solid #FFF;
  color: #FFF;
  font-size: 1.5rem;
  margin: 0 auto;
  padding: .75rem 0;
  text-decoration: none; }

.nurish-logo {
  margin: 25vh 0 5vh;
  text-align: center;
  width: 25rem; }

.collapse-logo {
  margin: 0 0 5vh;
  animation: move-logo 1s; }

@keyframes move-logo {
  0% {
    margin: 25vh 0 5vh; }
  100% {
    margin: 0 0 5vh; } }
