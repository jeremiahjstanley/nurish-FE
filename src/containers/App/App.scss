.App {
  text-align: center; 
}

.form-container {
  background-color: rgba(255, 255, 255, 0.2);
  border-radius: 30px;
  height: 25vh;
  margin: 0 auto;
  min-width: 20rem;
  padding: 1.5rem;
  text-align: center;
  transition: width 1 ease;
  width: 25%; 
}

.links {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
}

.links {
  & a {
    width: 75%;
  }
}

.link-container {
  display: flex;
  justify-content: space-between;
}

.link-container {
  & a {
    max-width: 10rem;
    width: 30%;
  }
}

a {
  border-radius: 30px;
  border: 1px solid #FFF;
  color: #FFF;
  font-size: 1.5rem;
  margin: 0 auto;
  padding: .75rem 0;
  text-decoration: none;
}

img {
  margin: 25vh 0 5vh;
  text-align: center; 
  width: 25rem;
}


