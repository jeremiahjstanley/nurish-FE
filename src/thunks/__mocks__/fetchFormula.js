export const fetchFormula = jest.fn();
export const fetchResults = jest.fn();