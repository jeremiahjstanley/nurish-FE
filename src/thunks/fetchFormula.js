import {isLoading, hasErrored, formulaFetchDataSuccess} from '../actions';
import {formulaCleaner} from '../data-helper/formula-cleaner.js';

export const fetchFormula = (url) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      if (!response.ok) {
        throw Error (response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(formulaFetchDataSuccess(result))
    } catch (error) {
      dispatch(hasErrored(true))
    }
  }
}