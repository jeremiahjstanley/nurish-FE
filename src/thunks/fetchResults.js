import {isLoading, hasErrored, filterResultFetchDataSuccess} from '../actions';
import {formulaCleaner} from '../data-helper/formula-cleaner.js';

export const fetchResults = (url) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      if (!response.ok) {
        throw Error (response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(filterResultFetchDataSuccess(result))
    } catch (error) {
      dispatch(hasErrored(true))
    }
  }
}