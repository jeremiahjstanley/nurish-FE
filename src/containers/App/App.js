import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter, NavLink, Route } from 'react-router-dom';
import Form from '../Form/Form';
import Search from '../Search/Search';
import FormulaContainer from '../FormulaContainer/FormulaContainer';
import Calculate from '../Calculate/Calculate';
import SingleCard from '../../components/SingleCard/SingleCard';
import Filter from '../Filter/Filter';
import './App.css';
import PropTypes from 'prop-types';
import { addFormulas } from '../../actions';
import { fetchFormula } from '../../thunks/fetchFormula.js'


class App extends Component {

  async componentDidMount() {
    const url = 'https://nurish-app.herokuapp.com/api/v1/formulas'
    await this.props.fetchFormula(url);
  }

  render() {
    return (
      <div className='app'>
        <img className='nurish-logo collapse-logo' src='/images/nurish-logo.gif'/>
        <Route exact path='/' component={Form}/>
        <Route exact path='/filter' component={Filter}/>
				<Route exact path='/search' component={Search}/>
				<Route exact path='/calculate' component={Calculate}/>
				<Route exact path='/browse' component={FormulaContainer}/>
        <Route exact path='/browse/:id' render={({ match }) => {
          const formula = formulas.find(formula => formula.id == match.params.id);
          return (
            <div>
              <SingleCard {...formula} />
            </div>
          );
        }}/>
      </div>
    );
  }
}


export const mapStateToProps = (state) => {
  formulas: state.formulas,
  isLoading: state.isLoading,
  hasErrored: state.hasErrored
}

export const mapDispatchToProps = (dispatch) => {
  addFormulas: (formulas) => dispatch(addFormulas(formulas)),
  fetchFormula: (url) => dispatch(fetchFormula(url))
}

App.proptypes = {
  formulas: PropTypes.array,
};
export default withRouter(connect(mapStateToProps, mapDsipatchToProps)(App));
