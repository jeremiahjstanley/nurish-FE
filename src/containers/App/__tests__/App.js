import {App, mapStateToProps, mapDispatchToProps} from '../App';
import React from 'react';
import { shallow } from 'enzyme';
import * as action from '../../actions';
import { fetchCheese } from '../../thunks/fetchFormula';

jest.mock('../../../thunks/fetchFormula');

describe('App', () => {
  let wrapper
  let mockFormulas
  let mockIsLoading
  let mockHasErrored
  let mockFetchFormula
  let mockAddFormula
})

  beforeEach(() => {
    mockFormulas = [{title: 'boost'}, { title: 'smoothie'}];
    mockFetchFormula = jest.fn();
    mockAddFormula = jest.fn();
    mockHasErrored = false
    mockIsLoading = true
  })


it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot()
  })