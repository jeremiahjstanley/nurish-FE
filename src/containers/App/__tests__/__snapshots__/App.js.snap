// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapShot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addFormulas={[Function]}
    fetchFormula={[Function]}
    formulas={
        Array [
            Object {
              "title": "boost",
            },
            Object {
              "title": "smoothie",
            },
          ]
    }
    hasErrored={false}
    isLoading={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          className="nurish-logo initial-logo"
          src="/images/nurish-logo.gif"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/filter"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/search"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/calculate"
/>,
        <Route
          exact={true}
          path="/formulas"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/formulas/:id"
          render={[Function]}
/>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "nurish-logo initial-logo",
          "src": "/images/nurish-logo.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/filter",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/search",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/calculate",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/formulas",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/formulas/:id",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="nurish-logo initial-logo"
            src="/images/nurish-logo.gif"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/filter"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/search"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/calculate"
/>,
          <Route
            exact={true}
            path="/formulas"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/formulas/:id"
            render={[Function]}
/>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "nurish-logo initial-logo",
            "src": "/images/nurish-logo.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/filter",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/search",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/calculate",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/formulas",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/formulas/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
